src/services/github.service.ts

```diff
@@ -21,111 +40,101 @@ export class GithubService {
       `${this.gitHubUrl}/repos/${username}/${repo}/branches`,
   };
 
-  public constructor(private httpService: HttpService) {}
+  getAllRepos(userName: string, headers: any): Observable<Repository[]> {
+    const headersForGitHub = this.headersBuilder.getHeadersForGitHub(headers);
+
+    return this.getUser(userName, headersForGitHub).pipe(
```

Comment:
This nesting does not look right, it should be sth more clear:
`this.getUser(userName, headersForGitHub).pipe(
mergeMap((user: [YourTypeHere]) => this.getNonForkRepose(users, headers))
`
and then mapping branches with repos should be part of getting the repos